libraries/botbuilder-dialogs/tests/componentDialog.test.js

```diff
@@ -347,8 +347,95 @@ describe('ComponentDialog', function () {
             .assertReply('Child finished.')
             .then(() => done());
     });
+
+    it('should handle that a components children have changed.', (done) => {
+        const conversationState = new ConversationState(new MemoryStorage());
+        const dialogState = conversationState.createProperty('dialog');
+
+        const childDialog = new WaterfallDialog('child', [
+            async step => {
+                component.addDialog(new WaterfallDialog('change'));
+                await step.context.sendActivity('First step.');
+                return Dialog.EndOfTurn;
+            },
+            async step => {
+                await step.context.sendActivity('Second step.');
+                return await step.endDialog();
+            }
+        ]);
+        const component = new ChangedDialog('test', true);
+        component.addDialog(childDialog);
+
+        const dialogs = new DialogSet(dialogState);
+        dialogs.add(component);
+
+        const adapter = new TestAdapter(async turnContext => {
+            const dc = await dialogs.createContext(turnContext);
+            const results = await dc.continueDialog();
+
+            if (results.status === DialogTurnStatus.empty) {
+                await dc.beginDialog('test');
+            } else {
+                assert(results.status === DialogTurnStatus.complete, `results.status should be 'complete' not ${ results.status }`);
+                assert(results.result === undefined, `results.result should be undefined, not ${ results.result }`);
```

Comment:
To avoid having to add descriptive error messages, this could be simplified to

```suggestion
                assert.strictEqual(results.status, DialogTurnStatus.complete);
                assert.strictEqual(results.result, undefined);
```